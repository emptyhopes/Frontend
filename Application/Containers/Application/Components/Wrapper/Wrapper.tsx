import styled from "styled-components";

import { WrapperInterface } from "@/Application/Containers/Application/Components/Wrapper/Types/WrapperInterface";

const StyledWrapper = styled.main`
  flex-grow: 1;
`;

const Wrapper = (props: WrapperInterface) => {
  return <StyledWrapper {...props}></StyledWrapper>;
};

export { Wrapper };
