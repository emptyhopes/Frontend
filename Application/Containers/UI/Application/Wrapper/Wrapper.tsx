import styled from "styled-components";

import { PropsInterface } from "@/Application/Containers/UI/Application/Wrapper/Types/PropsInterface";

const StyledWrapper = styled.main`
  flex-grow: 1;
  background-color: ${(props) => props.theme.header.background.color};
`;

const Wrapper: React.FunctionComponent<PropsInterface> = (props: PropsInterface): React.ReactElement => {
  return (
    <>
      <StyledWrapper {...props} />
    </>
  );
};

export { Wrapper };
