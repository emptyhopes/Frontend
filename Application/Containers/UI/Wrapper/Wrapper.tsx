import styled from "styled-components";

import { FunctionComponent } from "react";

import { WrapperInterface } from "@/Application/Containers/UI/Wrapper/Types/WrapperInterface";

const StyledWrapper = styled.main`
  flex-grow: 1;
`;

const Wrapper: FunctionComponent<WrapperInterface> = (props) => {
  return (
    <>
      <StyledWrapper {...props} />
    </>
  );
};

export { Wrapper };
