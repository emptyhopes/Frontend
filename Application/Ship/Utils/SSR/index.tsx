import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
