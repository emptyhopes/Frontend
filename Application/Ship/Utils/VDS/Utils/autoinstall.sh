#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
