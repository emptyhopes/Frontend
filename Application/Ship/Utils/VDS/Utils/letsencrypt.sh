#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname