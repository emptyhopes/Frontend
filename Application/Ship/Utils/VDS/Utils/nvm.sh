#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
