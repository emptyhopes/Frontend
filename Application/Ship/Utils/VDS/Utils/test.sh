#! /bin/bash

hostname="gartmann-art.ru"

curl --location -I http://$hostname.ru
curl --location -I http://en.$hostname.ru
curl --location -I http://ru.$hostname.ru

curl --location -I http://www.$hostname.ru
curl --location -I http://www.en.$hostname.ru
curl --location -I http://www.ru.$hostname.ru

curl --location -I https://$hostname.ru
curl --location -I https://en.$hostname.ru
curl --location -I https://ru.$hostname.ru

curl --location -I https://www.$hostname.ru
curl --location -I https://www.en.$hostname.ru
curl --location -I https://www.ru.$hostname.ru