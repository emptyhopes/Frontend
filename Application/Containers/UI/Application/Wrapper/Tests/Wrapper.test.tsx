import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { Wrapper } from "@/Application/Containers/UI/Application/Wrapper/Wrapper";

test("Wrapper", () => {
  const { container } = RenderTests(<Wrapper />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
