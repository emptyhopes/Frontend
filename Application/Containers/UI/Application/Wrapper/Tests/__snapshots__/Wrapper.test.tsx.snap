// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  background-color: #ffffff;
}

<div>
  <main
    class="c0"
  />
</div>
`;
